docusaurus/docs/dev-docs/configurations/typescript.md

```diff
@@ -16,3 +23,38 @@ description: How to configure Strapi for TypeScript development.
 | `tsconfig.json` file                      | application root | Manages TypeScript compilation for the server.                                                                                                    |
 | `tsconfig.json` file                      | `./src/admin/`   | Manages TypeScript compilation for the admin panel.                                                                                               |
 
+## Strapi-specific configuration for TypeScript <BetaBadge />
+
+When a Strapi project uses TypeScript, an optional `config/typescript.js|ts` configuration file can be added, which currently accepts only one parameter:
```

Comment:
You can use automatic type generation & IntelliSense even without a TS project. Those are two different things

Reply:
What are types in a non-TypeScript project, then? 🤔

Reply:
Types generated by Strapi are based on the user application structure and augment Strapi's own APIs (entity-service, etc...).
Thus, we can generate types for a JavaScript application as we're only going to look at the structure. Once the type definitions are emitted into their dedicated files, Strapi will read them to adapt its auto-completion & Intellisense accordingly. This works for JavaScript & TypeScript app.
The configuration file is called `typescript` as it configures **Strapi features** related to TypeScript like autogen. The TypeScript app configuration can be tweaked using the root tsconfig.json files.